tensorboard/components/tf_backend/router.ts

```diff
@@ -79,4 +86,63 @@ export function setRouter(router: Router): void {
   _router = router;
 }
 
+function addQueryParams(urlString: string, params?: QueryParams): string {
+  if (urlString.startsWith('/')) {
+    urlString = window.location.origin + urlString;
+  }
+  const url = new URL(urlString);
+  const searchParams = createSearchParam(params);
+  url.search = searchParams.toString();
+  return url.toString();
+}
+
+function createSearchParam(params: QueryParams = {}): URLSearchParams {
```

Comment:
Given that this is new functionality, it seems better to me to just require passing in an actual URLSearchParams object, rather than sticking with the custom QueryParams type.  You can construct URLSearchParams directly from a map, so most callsites can be updated by just wrapping a `new URLSearchParams()` around the argument.  I don't actually know that we have any callsites that use repeated/list-valued parameters, but if so, they can pass in a list of pairs instead or build up the instance via append().

This way the code in e.g. the audio plugin can be much simpler since there's no need to create URLSearchParams, convert it into QueryParams, and then reconvert back.